### PR TITLE
CBG-1273: Add more prometheus alert examples

### PR DESCRIPTION
**Pull request by JRascagneres**
Added a number of prometheus alert examples in our examples rule file.

### CODE REVIEW

File: examples/prometheus/rules/sync-gateway.rules.yml [hunk 1]
```diff
@@ -4,37 +4,86 @@ groups:
...
       - alert: HighChannelCacheMissRate
-        expr: sgw_cache_chan_cache_misses / (sgw_cache_chan_cache_misses + sgw_cache_chan_cache_hits) >= 1.0
+        expr: sgw_cache_chan_cache_misses / (sgw_cache_chan_cache_misses + sgw_cache_chan_cache_hits) >= 0.8
         for: 1m
         labels:
           severity: warning
         annotations:
           summary: High Channel Cache Miss Rate
+      - alert: HighDeltaCacheMissRate
+        expr: sgw_delta_sync_delta_sync_miss / (sgw_delta_sync_delta_sync_miss + sgw_delta_sync_delta_cache_hit) >= 0.8
```
Comment (adamcfraser) [inline examples/prometheus/rules/sync-gateway.rules.yml:42]: As discussed, confirm behaviour when stats aren't present (e.g. div by zero)

File: examples/prometheus/rules/sync-gateway.rules.yml [hunk 2]
```diff
@@ -4,37 +4,86 @@ groups:
...
+        annotations:
+          summary: An error occurred in the Last Hour
+      - alert: WarnChannelsPerDocCount
+        expr: increase(warn_channels_per_doc_count[1h]) > 0
+        for: 1m
+        labels:
+          severity: warning
+        annotations:
+          summary: A document attempted to exceed the channels per doc limit in the lsat hour
+      - alert: WarnGrantsPerDoc
```
Comment (adamcfraser) [inline examples/prometheus/rules/sync-gateway.rules.yml]: I don't think grants per doc is common enough to be included in examples.

File: examples/prometheus/rules/sync-gateway.rules.yml [hunk 3]
```diff
@@ -4,37 +4,86 @@ groups:
...
+        annotations:
+          summary: High Delta Cache Miss Rate
+      - alert: GlobalErrorCount
+        expr: increase(sgw_resource_utilization_error_count[1h]) > 1
+        for: 1m
+        labels:
+          severity: warning
+        annotations:
+          summary: An error occurred in the Last Hour
+      - alert: WarnChannelsPerDocCount
```
Comment (adamcfraser) [inline examples/prometheus/rules/sync-gateway.rules.yml]: Remove, needs review of the threshold for this warning.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: examples/prometheus/rules/sync-gateway.rules.yml (hunk 1 above)
Comment (JRascagneres) [reply examples/prometheus/rules/sync-gateway.rules.yml:42]: This is handled
- Absent stats are checked for and handled
- Divide by zero returns NaN as expected and doesn't trigger the alert

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**